src/models/hooks/objective.js

```diff
@@ -190,6 +192,65 @@ const propagateMetadataToTemplate = async (sequelize, instance, options) => {
   }
 };
 
+const updateParentGoalStatus = async (sequelize, instance) => {
+  const { goalId } = instance;
+  const goal = await sequelize.models.Goal.findOne({
+    where: {
+      id: goalId,
+      [Op.or]: [
+        {
+          // assumption: we only want to auto update the goal status if it is
+          // on an activity report, otherwise it should be updated by the
+          // RTR form logic or the RTR page
+          createdVia: 'activityReport',
+        },
+        {
+          onApprovedAR: true,
+        },
+      ],
+    },
+  });
+
+  if (goal && goal.status !== 'Closed') {
+    // here we will determine whether or not the goal should be updated
+    const objectives = await sequelize.models.Objective.findAll({
+      where: { goalId },
+    });
+
+    // is at least one obective not started?
+    const atLeastOneObjectiveIsNotStarted = objectives.some((objective) => objective.status === 'Not Started');
+
+    // is at least one objective in progress?
+    const atLeastOneObjectiveIsInProgress = objectives.some((objective) => objective.status === 'In Progress');
+
+    // are all objectives closed?
+    const allObjectivesAreClosed = objectives.every((objective) => objective.status === 'Completed');
+
+    // and so forth
+    let newStatus;
+
+    // we don't want any automatic regressions from "not started" to "in progress"
+    if (atLeastOneObjectiveIsNotStarted && goal.status !== 'In Progress') {
```

Comment:
Is there any concern here for overlapping true conditions? For example if more than one of these IF statements is true it might overwrite a previous one.

Or maybe this has already been taking into account given the ordering of the IF statements.

Reply:
yeah, that is the idea